app/src/main/java/in/ac/iitb/gymkhana/tumtumtracker/MapsActivity.java

```diff
@@ -37,10 +49,56 @@ protected void onCreate(Bundle savedInstanceState) {
     @Override
     public void onMapReady(GoogleMap googleMap) {
         mMap = googleMap;
-
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if (ContextCompat.checkSelfPermission(this,
+                    Manifest.permission.ACCESS_FINE_LOCATION)
+                    == PackageManager.PERMISSION_GRANTED) ;
+            else {
+                Toast toast = Toast.makeText(this, "Need Permission", Toast.LENGTH_SHORT);
+                toast.show();
+                requestPermissions(new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, 0);
+            }
+        }
         // Add a marker in Sydney and move the camera
-        LatLng sydney = new LatLng(-34, 151);
-        mMap.addMarker(new MarkerOptions().position(sydney).title("Marker in Sydney"));
-        mMap.moveCamera(CameraUpdateFactory.newLatLng(sydney));
+        // You need to change these values for current location
+        // LatLng sydney = new LatLng(-34, 151);
+        //mMap.addMarker(new MarkerOptions().position(sydney).title("Marker in Sydney"));
+        //mMap.moveCamera(CameraUpdateFactory.newLatLng(sydney));
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
This does not work as expected on my phone. I get a permission prompt and on clicking _Allow_, there are 2 consecutive toasts _Need Permission_ and _Restart after giving permission_ in that order